src/test/java/de/tum/cit/aet/artemis/nebula/NebulaFaqIntegrationTest.java

```diff
@@ -0,0 +1,87 @@
+package de.tum.cit.aet.artemis.nebula;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.cit.aet.artemis.core.domain.Course;
+import de.tum.cit.aet.artemis.core.user.util.UserUtilService;
+import de.tum.cit.aet.artemis.nebula.architecture.AbstractNebulaIntegrationTest;
+import de.tum.cit.aet.artemis.nebula.config.NebulaEnabled;
+import de.tum.cit.aet.artemis.nebula.dto.FaqConsistencyDTO;
+import de.tum.cit.aet.artemis.nebula.dto.FaqConsistencyResponse;
+import de.tum.cit.aet.artemis.nebula.dto.FaqRewritingDTO;
+import de.tum.cit.aet.artemis.nebula.dto.FaqRewritingResponse;
+import de.tum.cit.aet.artemis.nebula.service.FaqProcessingService;
+import de.tum.cit.aet.artemis.programming.util.ProgrammingExerciseUtilService;
+
+@Conditional(NebulaEnabled.class)
+class NebulaFaqIntegrationTest extends AbstractNebulaIntegrationTest {
```

Comment:
You should probably add more tests here to also cover edge cases. (e.g. testing the error handling)